docs/public-networks/concepts/parallel-transaction-execution.md

```diff
@@ -0,0 +1,191 @@
+---
+sidebar_position: 4
+description: Learn about parallel transaction execution.
+tags:
+- public networks
+---
+
+# Parallel transaction execution
+
+Besu supports parallel transaction execution, using an optimistic approach to parallelize
+transactions within a block.
+You can enable this feature when using the [Bonsai Tries](data-storage-formats.md#bonsai-tries) data
+storage format.
+This page provides an [overview of the parallelization mechanism](#parallelization-mechanism-overview),
+and [metrics](#metrics) that highlight Besu's improved performance.
+
+:::warning Important
+Parallel transaction execution is an early access feature.
+You can enable it using the `--Xbonsai-parallel-tx-processing-enabled` option.
+:::
+
+## Parallelization mechanism overview
+
+When parallel transaction execution is enabled, Besu initially executes all transactions within a
+block in parallel, operating under the optimistic assumption that they can all be executed
+concurrently without conflict.
+This parallel execution runs in the background, and Besu proceeds to sequentially process the
+transactions without waiting for the parallel execution to complete.
+
+The following flowchart outlines the transaction execution flow:
+
+<p align="center">
+
+```mermaid
+graph TD;
+  X(Start parallel execution as background process) --> A(Start sequential processing);
+  A --> B{{Is transaction finalized?}};
+  B --> |Yes| C{{Conflict check}};
+  C --> |No conflict| D(Apply background state modifications);
+  C --> |Conflict detected| E(Replay transaction using background cache);
+  B --> |No| F(Execute transaction sequentially);
+  D --> G(End sequential processing);
+  E --> G;
+  F --> G;
+```
+
+</p>
+
+Besu first determines if a transaction has been "finalized," or completed by the background parallel
+execution:
+
+- **Finalized:** If the transaction is finalized, Besu examines whether there are any conflicts with
```

Comment:
I would use completed instead of finalized in this context, as we may think that it is related to the concept of "finalized" in PoS context,